source/LoadingPanel.cpp

```diff
@@ -0,0 +1,90 @@
+/* LoadingPanel.cpp
+Copyright (c) 2021 by Michael Zahniser
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#include "LoadingPanel.h"
+
+#include "Angle.h"
+#include "Audio.h"
+#include "GameData.h"
+#include "Interface.h"
+#include "Information.h"
+#include "MenuPanel.h"
+#include "MaskManager.h"
+#include "PlayerInfo.h"
+#include "Point.h"
+#include "PointerShader.h"
+#include "Ship.h"
+#include "SpriteSet.h"
+#include "StarField.h"
+#include "System.h"
+#include "UI.h"
+
+#include "gl_header.h"
+
+using namespace std;
+
+
+
+LoadingPanel::LoadingPanel(PlayerInfo &player, UI &gamePanels)
+	: player(player), gamePanels(gamePanels)
+{
+	SetIsFullScreen(true);
+}
+
+
+
+void LoadingPanel::Step()
+{
+	progress = static_cast<int>(GameData::Progress() * 60.);
+
+	// If the game isn't loaded yet, upload any images to the GPU.
+	GameData::ProcessSprites();
+
+	if(progress == 60)
+	{
+		// Now that we have finished loading all the basic sprites and sounds, we can look for invalid file paths,
+		// e.g. due to capitalization errors or other typos.
+		SpriteSet::CheckReferences();
+		Audio::CheckReferences();
+		// All sprites with collision masks should also have their 1x scaled versions, so create
+		// any additional scaled masks from the default one.
+		GameData::GetMaskManager().ScaleMasks();
+
+		GameData::FinishLoading();
+
+		player.LoadRecent();
```

Comment:
Is it possible that this player.LoadRecent() call happens while a Test already started? (This call clears all data in player.)

Reply:
I created PR https://github.com/quyykk/endless-sky/pull/1 which appears to solve the issue for the `"Conditions Arithmetic and Loops"` test. The fix is not very elegant, I hope you can come up with something better.

Reply:
You're right. I only checked to see if this is called multiple times. Thank you!

Reply:
I fixed it by given `LoadingPanel` the authority to say when the data has loaded.

Reply:
I also have some fixes for the other integration test in https://github.com/quyykk/endless-sky/pull/2. That PR is just for merging directly.

Reply:
Thank you